nav2_controller/src/nav2_controller.cpp

```diff
@@ -63,6 +70,15 @@ ControllerServer::on_configure(const rclcpp_lifecycle::State & state)
 {
   RCLCPP_INFO(get_logger(), "Configuring controller interface");
 
+  get_parameter("progress_checker_plugin", progress_checker_id_);
+  if (progress_checker_id_ == default_progress_checker_id_) {
+    declare_parameter(default_progress_checker_id_ + ".plugin", default_progress_checker_type_);
+  }
+  get_parameter("goal_checker_plugin", goal_checker_id_);
+  if (goal_checker_id_ == default_goal_checker_id_) {
+    declare_parameter(default_goal_checker_id_ + ".plugin", default_goal_checker_type_);
```

Comment:
I think that might be missing.

Reply:
Do we have to mention these? The user does not need to declare them as that will be done in nav2_util::get_plugin_type_param automatically.

Reply:
The declare here needs to be in the constructor or changed to a declare if not already declared or else multiple configurations will crash (currently dealing with that for the existing plugins that make that mistake)

Reply:
Done